docs/source/en/tasks/object_detection.mdx

```diff
@@ -0,0 +1,487 @@
+            "bbox": objects["bbox"][i],
+        }
+        annotations.append(new_ann)
+
+    return annotations
+
+
+# CocoDetection expects to read the data from files in a certain format
+def save_cppe5_annotation_file_images(cppe5):
+    output_json = {}
+    path_output_cppe5 = f"{os.getcwd()}/cppe5/"
+
+    if not os.path.exists(path_output_cppe5):
+        os.makedirs(path_output_cppe5)
+
+    path_anno = os.path.join(path_output_cppe5, "cppe5_ann.json")
+    categories_json = [{"supercategory": "none", "id": id, "name": id2label[id]} for id in id2label]
+    output_json["images"] = []
+    output_json["annotations"] = []
+    for batch in cppe5:
+        ann = val_formatted_anns(batch["image_id"], batch["objects"])
+        output_json["images"].append(
+            {
+                "id": batch["image_id"],
+                "width": batch["image"].width,
+                "height": batch["image"].height,
+                "file_name": f"{batch['image_id']}.png",
+            }
+        )
+        output_json["annotations"].extend(ann)
+    output_json["categories"] = categories_json
+
+    with open(path_anno, "w") as file:
+        json.dump(output_json, file, ensure_ascii=False, indent=4)
+
+    for im, img_id in zip(cppe5["image"], cppe5["image_id"]):
+        path_img = os.path.join(path_output_cppe5, f"{img_id}.png")
+        im.save(path_img)
+
+    return path_output_cppe5, path_anno
+
+
+path_output_cppe5, path_anno = save_cppe5_annotation_file_images(cppe5["test"])
+
+dummy = CocoDetection(path_output_cppe5, im_processor, path_anno)
+module = evaluate.load("ybelkada/cocoevaluate", coco=dummy.coco)
+val_dataloader = torch.utils.data.DataLoader(dummy, batch_size=8, shuffle=False, num_workers=4, collate_fn=collate_fn)
+
+with torch.no_grad():
+    for idx, batch in enumerate(tqdm(val_dataloader)):
+        pixel_values = batch["pixel_values"]
+        pixel_mask = batch["pixel_mask"]
+
+        labels = [
+            {k: v for k, v in t.items()} for t in batch["labels"]
+        ]  # these are in DETR format, resized + normalized
+
+        # forward pass
+        outputs = model(pixel_values=pixel_values, pixel_mask=pixel_mask)
+
+        orig_target_sizes = torch.stack([target["orig_size"] for target in labels], dim=0)
+        results = im_processor.post_process(outputs, orig_target_sizes)  # convert outputs of model to COCO api
+
+        module.add(prediction=results, reference=labels)
+        del batch
+
+results = module.compute()
+print(results)
+```
+
+## Inference
+Now that you have finetuned a DETR model, evaluated it and uploaded it to the 🤗Hub, you can use it for inference.
+
+```py
+>>> import requests
+
+>>> url = "https://chartwell.com/-/media/Images/blog/2020/04/keeping-staff-and-residents-protected.jpeg?mw=1900&hash=B8A76374D6BD24B0DC8A779D0C985027"
+>>> image = Image.open(requests.get(url, stream=True).raw)
+
+>>> image_processor = DetrImageProcessor.from_pretrained("MariaK/detr-resnet-50_fine_tuned_cppe5")
```

Comment:
The first example should just use the `pipeline` function, which will take care of all pre/post processing. Then you can dive into doing it by hand.

Reply:
Added an example of inference with the pipeline. Fixed in 1a38260b27ef977e0d82ba1c918eeb19c8f3dcb7